### DISCUSSION TITLE
SyntaxError: Cannot use import statement outside a module

### DISCUSSION THREAD

**Discussion by inovramadani** [babel/babel#12463] [Q&A] [answered]
## Bug Report

**Current behavior**
I faced intricate issue I think it's related to babel, posted on SOF: https://stackoverflow.com/questions/65164157/set-up-testing-library-react-with-mocha-got-syntaxerror-unexpected-token

Please clone this repo: https://github.com/inovramadani/babel-bug
And then run `npm test`
Should get this error:
```
import Button from './button';
^^^^^^

SyntaxError: Cannot use import statement outside a module
```

**Expected behavior**
Should be able to run the test smoothly and not getting any error.

**Environment**
Tried on 3 different OS-es and 3 different npm and node versions, got same error. So I think has nothing to do with the environment.

[a comment by an automated account is not shown]

**Comment by inovramadani** [comment]
hi @babel-bot, the sign up link is expired:  
![image](https://user-images.githubusercontent.com/27492820/101605019-41b4a400-3a34-11eb-9782-f2e3540a620a.png)

**Comment by nicolo-ribaudo** [comment, marked as answer]
You need to add `"sourceType": "module"` to your Babel config.

**Reply by inovramadani** [reply]
@nicolo-ribaudo did you clone the repo and get it fixed by adding `"sourceType": "module"` on .babelrc file? I tried on mine, the error still appears.

**Reply by nicolo-ribaudo** [reply]
Oh I tested it on CodeSandbox and for some reason it was marking all the tests as passing.

The error you are getting is from a dependency. By default, `@babel/register` does not compile `node_modules`. You can make it compile everything by explicitly setting an `only` or `ignore` option.

First, you need to rename `.babelrc` to `babel.config.json` so that its plugins can be applied to other packages ([docs](https://babeljs.io/docs/en/config-files)).
Then, you need to create a file (for example, `babel-register.js`) where you configure `@babel/register`:
```js
require("@babel/register")({
    only: [/\.jsx?/]
});
```
Lastly, you use this file instead of `@babel/register` in `.mocharc.yml`.

With this, I got past that error. It's now throwing about because you are doing `import "style.css"` and that's (obviously) not valid JS. You'll need somehow to configure mocha with an additional tool (not Babel) that lets it load CSS files.

**Reply by Alexzanderk** [reply]
Who can create a boilerplate with all of this and paste the link for it?

**Reply by Angelk90** [reply]
@nicolo-ribaudo : Who can create a boilerplate with all of this and paste the link for it?

**Reply by aniketsingh98571** [reply]
const kol=require('./Basics')
